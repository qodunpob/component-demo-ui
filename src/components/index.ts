export * from './gitHubLink'
export * from './preview'
export * from './propList'
export * from './section'
