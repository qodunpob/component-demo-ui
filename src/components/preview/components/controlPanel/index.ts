export * from './ControlPanel'
export * from './ControlPanel.types'
