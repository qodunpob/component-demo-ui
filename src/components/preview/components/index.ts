export * from './controlPanel'
export * from './inspector'
