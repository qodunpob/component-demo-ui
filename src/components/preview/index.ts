export * from './Preview'
export * from './Preview.types'
export type { ControlDefinition } from './components'
