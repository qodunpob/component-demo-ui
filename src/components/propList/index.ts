export * from './PropList'
export * from './PropList.types'
