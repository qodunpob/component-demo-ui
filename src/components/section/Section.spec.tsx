import React from 'react'
import { render, type RenderResult } from '@testing-library/react'
import { Section, type SectionProps } from '~/components'

describe('Section', () => {
  it('should display the title and content of the section', () => {
    const { baseElement } = renderComponent()
    expect(baseElement).toMatchSnapshot()
  })
})

const renderComponent = ({
  title = 'Section',
  ...otherProps
}: Partial<SectionProps> = {}): RenderResult =>
  render(
    <Section title={title} {...otherProps}>
      <p>
        Here is the <b>section content</b>
      </p>
    </Section>
  )
