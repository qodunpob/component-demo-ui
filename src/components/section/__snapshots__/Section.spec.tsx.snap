// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section should display the title and content of the section 1`] = `
<body>
  <div>
    <div
      class="component-demo-ui-container component-demo-ui-section"
    >
      <h2
        class="component-demo-ui-section-title"
      >
        Section
      </h2>
      <p>
        Here is the 
        <b>
          section content
        </b>
      </p>
    </div>
  </div>
</body>
`;
